pkg/apis/policy/types.go

```diff
@@ -444,3 +449,18 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy
 }
+
+// RuntimeClassStrategyOptions define the strategy that will dictate the allowable RuntimeClasses
+// for a pod.
+type RuntimeClassStrategyOptions struct {
+	// AllowedRuntimeClasses is a whitelist of RuntimeClass names that may be specified on a pod.
+	// An empty string value allows the RuntimeClassName to be left unset (or set to "").
+	// An empty list allows all RuntimeClasses.
+	// +optional
+	AllowedRuntimeClasses []string
```

Comment:
**Question:** Should this be a label selector on RuntimeClasses, rather than a whitelist?

A whitelist gives control to the PodSecurityPolicy owner, and requires the policy be kept up to date as new classes are added. A label selector gives the RuntimeClass creator control, and requires that all RuntimeClasses have the necessary labels.

Reply:
This is an interesting question because this is the first time the PSP is governing a reference to another object. I think we should stick with the whitelist approach for now, because otherwise the PSP needs to start looking up RuntimeClasses in order to check labels, also because we expect the overall number of RuntimeClasses to be relatively small and unchanging.